File: clojure/allergies/src/allergies.clj
(ns allergies)

(def ^:private allergens
  [:eggs :peanuts :shellfish :strawberries :tomatoes :chocolate :pollen :cats])

(defn allergies [score]
  (->> (range (count allergens))
       (filter (partial bit-test score))
       (map (partial get allergens))))

(defn allergic-to? [score allergen]
  (bit-test score (.indexOf allergens allergen)))

File: ruby/allergies/allergies.rb
class Allergies
  ALLERGENS = %w[eggs peanuts shellfish strawberries tomatoes chocolate pollen cats]

  attr_reader :score

  def initialize(score)
    @score = score
  end

  def allergic_to?(allergen)
    score & (1 << ALLERGENS.index(allergen)) != 0
  end

  def list
    ALLERGENS.filter { allergic_to? _1 }
  end
end
